tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPMirrorConnectionTest.java

```diff
@@ -229,6 +271,173 @@ public void testBrokerAddsAddressAndQueue() throws Exception {
+            .withName(Matchers.startsWith("$ACTIVEMQ_ARTEMIS_MIRROR"))
+            .withDesiredCapabilities(capabilities)
+            .respond()
+            .withOfferedCapabilities(capabilities)
+            .withPropertiesMap(brokerProperties);
+         peer_3.remoteFlow().withLinkCredit(10).queue();
+         peer_3.start();
+
+         final URI remoteURI = peer_3.getServerURI();
+         logger.info("Connect test started, peer listening on: {}", remoteURI);
+
+         final int AMQP_PORT_2 = BROKER_PORT_NUM + 1;
+         final ActiveMQServer server_2 = createServer(AMQP_PORT_2, false);
+         {
+            AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "toPeer3", "tcp://" + remoteURI.getHost() + ":" + remoteURI.getPort());
+            amqpConnection.setReconnectAttempts(0);// No reconnects
+            amqpConnection.setUser("user");
+            amqpConnection.setPassword("pass");
+            amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement().addProperty(TUNNEL_CORE_MESSAGES, Boolean.toString(tunneling)).setQueueCreation(true).setAddressFilter(getQueueName() + ",sometest"));
+            server_2.getConfiguration().addAMQPConnection(amqpConnection);
+
+            amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "toServer", "tcp://localhost:" + BROKER_PORT_NUM);
+            amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement().setAddressFilter(getQueueName()));
+            server_2.getConfiguration().addAMQPConnection(amqpConnection);
+         }
+
+         {
+            AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "toServer2", "tcp://localhost:" + AMQP_PORT_2).setRetryInterval(50);
+            amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement().setAddressFilter(getQueueName()).setNoForward(true));
+            server.getConfiguration().addAMQPConnection(amqpConnection);
+         }
+
+         // connect the topology
+         server_2.start();
+         server.start();
+         peer_3.waitForScriptToComplete(5, TimeUnit.SECONDS);
+
+         // Create queues & send messages on server, nothing will reach peer_3
+         createAddressAndQueues(server);
+         Wait.assertTrue(() -> server_2.locateQueue(getQueueName()) != null);
+         Wait.assertTrue(() -> server.locateQueue(getQueueName()) != null);
+
+         final org.apache.activemq.artemis.core.server.Queue q1 = server.locateQueue(getQueueName());
+         assertNotNull(q1);
+
+         final org.apache.activemq.artemis.core.server.Queue q2 = server_2.locateQueue(getQueueName());
+         assertNotNull(q2);
+
+         final ConnectionFactory factory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:" + BROKER_PORT_NUM);
+         final ConnectionFactory factory_2 = CFUtil.createConnectionFactory("CORE", "tcp://localhost:" + AMQP_PORT_2);
```

Comment:
Probably worth renaming the test to make clear it is Core large messages being sent originally. Equally, if controlling the AMQP large message size (which here, will only matter for the non-tunnelled case as mirrored messages arrive at server 2 via server 1) then it probably makes sense to configure the Core client large message size (via its URI, e.g add "?minLargeMessageSize=\<bytes-size\>"

Probably worth having tests with large AMQP messages also.